ssdutil/main.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python
+#
+# main.py
+#
+# Command-line utility for interacting with ssd in SONiC
+#
+
+
+
+try:
+    from ssdutil import SsdUtil
+    import imp
+    import syslog
+    import click
+    import os
+    import sys
+except ImportError as e:
+    raise ImportError("%s - required module not found" % str(e))
+
+VERSION = '1.0'
+
+VENDOR_PATH = "ssdvendors"
+
+
+SYSLOG_IDENTIFIER = "ssdutil"
+
+# Global ssd vendor class instance
+global ssd_vendor
+
+
+# ========================== Syslog wrappers ==========================
+
+
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        click.echo(msg)
+
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        click.echo(msg)
+
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        click.echo(msg)
+
+# load ssd vendor module
+def load_ssd_vendor(device):
+    global ssd_vendor 
+    dvModel = SsdUtil().get_device_model(device)   
+    if "InnoDisk" in dvModel:    
+        try:
+            module_path = os.path.dirname(os.path.abspath(__file__))
+            module_file = "/".join([module_path, VENDOR_PATH, "innodisk.py"]) 
+            module = imp.load_source("innodisk", module_file)
+        except IOError, e:
+            log_error("Failed to load VENDOR module '%s': %s" % ("innodisk" , str(e)), True)
+            return -1        
+        try:
+            ssd_vendor_class = getattr(module, "InnoDisk")
+            ssd_vendor = ssd_vendor_class()
+        except AttributeError, e:
+            log_error("Failed to instantiate '%s' class: %s" % ("InnoDisk", str(e)), True)
+            return -2
+    elif "SanDisk" in dvModel:
```

Comment:
Hi Cary,here is for others to extend.Currently SanDisk is just for our test right now.Not the official release code.Please delete this part and detail the comment that other vendors can extend here.